src/EnergyPlus/WaterThermalTanks.hh

```diff
@@ -69,86 +69,185 @@ struct EnergyPlusData;
 
 namespace WaterThermalTanks {
 
-    enum struct AmbientTempEnum
+    enum class WTTAmbientTemp
     {
+        Unassigned = -1,
         Schedule,   // ambient temperature around tank (or HPWH inlet air) is scheduled
         TempZone,   // tank is located in a zone or HPWH inlet air is zone air only
         OutsideAir, // tank is located outdoors or HPWH inlet air is outdoor air only
-        ZoneAndOA   // applicable to HPWH only, inlet air is mixture of OA and zone air
+        ZoneAndOA,  // applicable to HPWH only, inlet air is mixture of OA and zone air
+        Num
     };
 
-    enum struct CrankTempEnum
+    constexpr std::array<std::string_view, static_cast<int>(WTTAmbientTemp::Num)> HPWHAmbientTempNamesUC{
+        "SCHEDULE", "ZONEAIRONLY", "OUTDOORAIRONLY", "ZONEANDOUTDOORAIR"};
+
+    constexpr int TankAmbientTempNamesNum = static_cast<int>(WTTAmbientTemp::Num) - 1; // Since AmbientTemp::ZoneAndOA is not appilcable to Tank
+    constexpr std::array<std::string_view, TankAmbientTempNamesNum> TankAmbientTempNamesUC{
+        "SCHEDULE",
+        "ZONE",
+        "OUTDOORS",
+    };
+
+    enum class CrankcaseHeaterControlTemp
     {
+        Unassigned = -1,
         Schedule, // temperature controlling compressor crankcase heater is scheduled
         Zone,     // temperature controlling compressor crankcase heater is zone air
-        Exterior  // temperature controlling compressor crankcase heater is outdoor air
+        Exterior, // temperature controlling compressor crankcase heater is outdoor air
+        Num
+    };
+
+    constexpr std::array<std::string_view, static_cast<int>(CrankcaseHeaterControlTemp::Num)> CrankcaseHeaterControlTempNamesUC{
+        "SCHEDULE",
+        "ZONE",
+        "OUTDOORS",
```

Comment:
Cut-n-Paste error? Should be EXTERIOR

Reply:
@jmythms can you respond to this?

Reply:
[Currently, Develop](HPWH.CrankcaseTempIndicator = CrankTempEnum::Exterior;) checks against the text `"OUTDOORS"` to set 
`HPWH.CrankcaseTempIndicator = CrankTempEnum::Exterior;`. I don't know if this is a bug... But the code in this PR has the same behavior as Develop, as of now.

Reply:
Checking the IDD file to confirm.

Reply:
One thought I had was that it doesn't matter what the string is so this still looks OK. Another thought is for the future where the string does not closely resemble the enum name. I could go either way here.

Reply:
I don't know how to link the specific line in the IDD file here (Github says the file is too big), but it is line number 78293, WaterHeater:HeatPump:PumpedCondenser object, field Compressor Location. The three keys are Schedule, Zone and Outdoors...
![image](https://user-images.githubusercontent.com/45446967/142024143-ddc272e3-1b79-49a8-9dbd-4ff17f18bd04.png)

Reply:
I agree it could be better if the string was the same as the enum name. Shall I change it?

Reply:
Change the enum name given what the IDD shows. Also, there are now 2 string arrays with the same strings, this one and one before this one.

Reply:
Thanks, will do!

Reply:
I don't know if I should combine them and just re-use one string array. This might save a tiny bit of memory, and both seem to be referring to a somewhat similar physical phenomenon: ambient air temperatures. On the other hand, it might be better to keep them separate for understandability: one refers to the tank ambient temperature and another to compressor ambient temperature.

Reply:
Yeah, just leave that for now.

Reply:
Thank you!